### DISCUSSION TITLE
Blurry FlxText on HTML5

### DISCUSSION THREAD

**Discussion by 13rice** [HaxeFlixel/flixel#2572] [Q&A]
Hello,

I'm targeting Html5 since the end of life of Flash, and I noticed that the text is blurry, even for Pixelated font with the recommended size. Example [this one](https://www.dafont.com/retroville-nc.font) with size 10 or 20.
I see the same issue on the Demo Code FlxTextFormat.
![text blurry issue](https://user-images.githubusercontent.com/16978288/171206259-318bfdd2-d5d1-4c20-8e19-32bcbd19eaed.png)

When I add a shadow it gets worse unfortunately. it was very clean before on Flash target.
I'm using HaxeFlixel 4.10 with OpenFL 9.1.0

**Flash**
![2020-04-26 flash text](https://user-images.githubusercontent.com/16978288/171206263-ffe276f4-bfb8-4e75-87c0-9bead19e80bb.png)

**Html5**
![2021-08-17 blurry html5](https://user-images.githubusercontent.com/16978288/171206268-f5db5f1f-b2b3-4d22-a920-28578a906fb2.png)

Is there a way to have pixel perfect font AND text formatting ?

I see Bitmap font but it looks more complicated to have different color, format etc.

**Comment by Geokureli** [comment]
We don't really have a perfect solution for this, the problem lies with rendering openfl text to an image, and it can be a hassle especially at different scales. I usually suggest FlxBitmapText over FlxText. to apply different colors you can use a white (or greyscale) bitmap and set either `text.color = FlxColor.RED` , or you can use `text.setColorTransform()`
